src/core/done.js

```diff
@@ -1,90 +1,41 @@
 'use strict'
 
-var groupBy = require( 'lodash.groupby' )
-var columnify = require( 'columnify' )
+var groupBy = require( 'lodash' ).groupBy
 
-function shouldExit1( warningsOrErrors ) {
-	var maxErrs = typeof this.config.maxErrors === 'number' ? this.config.maxErrors : null
-	var maxWarnings = typeof this.config.maxWarnings === 'number' ? this.config.maxWarnings : null
+function shouldExit1( warningsOrErrors, maxErrors, maxWarnings ) {
 	var numberofErrors = warningsOrErrors.Error && warningsOrErrors.Error.length || 0
 	var numberofWarnings = warningsOrErrors.Warning && warningsOrErrors.Warning.length || 0
 
-	if ( maxErrs == null && numberofErrors > 0 ) {
+	if ( maxErrors < 0 && numberofErrors > 0 ) {
 		return true
 	}
 
-	return maxErrs != null && numberofErrors > this.config.maxErrors ||
-		maxWarnings != null && numberofWarnings > this.config.maxWarnings
+	return maxErrors >= 0 && numberofErrors > maxErrors ||
+		maxWarnings >= 0 && numberofWarnings > maxWarnings
 }
 
 /**
  * @description outputs our messages, wipes errs/warnings if watching
  * @returns {Object | Function} returns process exit if not watching, or obj otherwise
  */
 var done = function() {
-	var msg = ''
-	var groupedByFile = {}
-	var msgGrouped
-	var group = this.config.groupOutputByFile
-	var opts = this.config.reporterOptions || {}
+	var maxErrors = typeof this.config.maxErrors === 'number' ? this.config.maxErrors : -1
+	var maxWarnings = typeof this.config.maxWarnings === 'number' ? this.config.maxWarnings : -1
 
-	var violations = this.cache.violations
-		.filter( function( msg ) { return !!msg && !!msg.message } )
-
-	var shouldKill = shouldExit1.call( this, groupBy( violations, 'severity' ) )
+	var shouldKill = shouldExit1( groupBy( this.cache.messages, 'severity' ), maxErrors, maxWarnings )
 
 	this.state.exitCode = shouldKill ? 1 : 0
 
-	// when testing we want to silence the console a bit, so we have the quiet option
-	if ( !this.state.quiet ) {
-		violations.forEach( function( msg ) { this.reporter( msg ) }.bind( this ) )
-
-		this.reporter( null, 'done', shouldKill ? 'kill' : null )
-
-		// by default group warnings and errs by file
-		if ( group && this.cache.messages ) {
-			this.cache.messages.forEach( function( output ) {
-				var file = output.file
-
-				if ( groupedByFile.hasOwnProperty( file ) ) {
-					groupedByFile[file].push( output )
-				}
-				else {
-					groupedByFile[file] = [output]
-				}
-			} )
+	var message = this.reporter( this.cache.messages, shouldKill, {
```

Comment:
Why some options are inside `options` object and some on their own as arguments?